webapp/src/routes/IncomeReport/IncomePieChart.js

```diff
@@ -0,0 +1,141 @@
+import React, { memo, useState } from 'react'
+import PropTypes from 'prop-types'
+import { makeStyles } from '@mui/styles'
+import { PieChart, Pie, Sector, ResponsiveContainer } from 'recharts'
+
+import styles from './styles'
+
+const useStyles = makeStyles(styles)
+
+let GLOBAL_COIN = ''
+
+const thousandSeparator = number => {
+  return number.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
+}
+
+const renderActiveShape = props => {
+  const RADIAN = Math.PI / 180
+  const {
+    cx,
+    cy,
+    midAngle,
+    innerRadius,
+    outerRadius,
+    startAngle,
+    endAngle,
+    fill,
+    payload,
+    percent,
+    value
+  } = props
+  const sin = Math.sin(-RADIAN * midAngle)
+  const cos = Math.cos(-RADIAN * midAngle)
+  const sx = cx + (outerRadius + 10) * cos
+  const sy = cy + (outerRadius + 10) * sin
+  const mx = cx + (outerRadius + 30) * cos
+  const my = cy + (outerRadius + 30) * sin
+  const ex = mx + (cos >= 0 ? 1 : -1) * 22
+  const ey = my
+  const textAnchor = cos >= 0 ? 'start' : 'end'
+
+  return (
+    <g>
+      <text x={cx} y={cy} dy={8} textAnchor="middle" fill={fill}>
+        {payload.name}
+      </text>
+      <Sector
+        cx={cx}
+        cy={cy}
+        innerRadius={innerRadius}
+        outerRadius={outerRadius}
+        startAngle={startAngle}
+        endAngle={endAngle}
+        fill={fill}
+      />
+      <Sector
+        cx={cx}
+        cy={cy}
+        startAngle={startAngle}
+        endAngle={endAngle}
+        innerRadius={outerRadius + 6}
+        outerRadius={outerRadius + 10}
+        fill={fill}
+      />
+      <path
+        d={`M${sx},${sy}L${mx},${my}L${ex},${ey}`}
+        stroke={fill}
+        fill="none"
+      />
+      <circle cx={ex} cy={ey} r={2} fill={fill} stroke="none" />
+      <text
+        x={ex + (cos >= 0 ? 1 : -1) * 12}
+        y={ey}
+        textAnchor={textAnchor}
+        fill="#333"
+      >{`${GLOBAL_COIN}-${thousandSeparator(Number(value))}`}</text>
+      <text
+        x={ex + (cos >= 0 ? 1 : -1) * 12}
+        y={ey}
+        dy={18}
+        textAnchor={textAnchor}
+        fill="#999"
+      >
+        {`(Rate ${(percent * 100).toFixed(2)}%)`}
+      </text>
+    </g>
+  )
+}
+
+const IncomePieChart = ({ data, coinType }) => {
+  const classes = useStyles()
+
+  GLOBAL_COIN = coinType
+
+  const [activeIndex, setActiveIndex] = useState(0)
+
```

Comment:
remove blank

Reply:
actually, it's ok this blank space, because the next const is a function BUT I would do some changes in order to reorder the lines.

example:
Move line 94 to line 91 and leave a blank space between variables declarations and the function declaration

```
 const classes = useStyles()
 const [activeIndex, setActiveIndex] = useState(0)

 GLOBAL_COIN = coinType
 
 const onPieEnter = (_, index) => {...
  ```